File: python_programming/chapter5/daddy.rb
pairs = {"Jon" => "Chris", "Chris" => "Ed", "Mark" => "Chris", "Jordan" => "Mark", "Steven" => "David"}

choice = ""

while choice != "0"


	print(
	"""

		0 - Quit
		1 - Look Up Pair
		2 - Add Pair
		3 - Delete Pair

	"""

	)

 
	print("Choice: ")
	choice = gets().chomp()

	# exit
	if choice == "0"
		print("Good-bye!")
	# lookup a pair
	elsif choice == "1"
		print("Enter son: ")
		son = gets().chomp()
		if pairs.include?(son)
			print(pairs[son])
		else
			print("#{son} not in dictionary")
		end
	
	# add a pair
	elsif choice == "2"
		print("Enter son to add: ")
		son = gets().chomp()
		if pairs.include?(son)
			print("#{son} is already in the dictionary.")
		else
			print("Enter the father of the son: ")
			father = gets().chomp()
			pairs[son] = father

		end
		
	# delete a pair
	elsif choice == "3"
		print("Enter a son to delete: ")
		son = gets().chomp()
		if pairs.include?(son)
			pairs.delete(son)
			print("#{son} deleted")
			
		else
			print("#{son} is already in the dictionary.")
		end
				
	end
end

File: python_programming/chapter5/daddy.py
pairs = {"Jon" : "Chris", "Chris" : "Ed", "Mark" : "Chris", "Jordan" : "Mark", "Steven" : "David"}

choice = None

while choice != "0":


	print(
	"""

		0 - Quit
		1 - Look Up Pair
		2 - Add Pair
		3 - Delete Pair

	"""

		)

	choice = input("Choice: ")
	# exit
	if choice == "0":
		print("Good-bye!")
	
	# look up pair
	if choice == "1":
		son = input("Enter the name of a son: ")
		if son in pairs:
			print(pairs[son])
		else:
			print(son, "is not in dictionary.")

	# add a pair
	if choice == "2":
		son = input("Enter the name of a son to add: ")
		if son not in pairs:
			father = input("Enter the name of the son's father: ")
			pairs[son] = father
		else:
			print(son, "already in dictionary.")

	# delete a pair
	if choice == "3":
		son = input("Enter name of son to delete: ")
		del pairs[son]
		print(son, "deleted.")

